config-no-istio/100-namespace.yaml

```diff
@@ -0,0 +1,20 @@
+# Copyright 2019 The Knative Authors
```

Comment:
Can you comment on the diff of all of these `config-no-istio` files with their `config` counterparts? This seems like a lot of duplication to me. Also `config-no-istio` should be `config-gloo`, shouldn't it?

Reply:
I agree that there's a lot of duplication in `config-no-istio` (which i'll rename as suggested). The challenge here is how to modify the existing `config` dir which has dependencies on Istio that aren't needed for the Gloo tests. the changes in the `config-no-istio` dir include:
- removing any references to istio crds
- removing any references to the `istio-system` namespace

Without some kind of chart templating system like Helm, it's hard to reuse the original manifest. I agree that a modified duplicate of the `config` dir is unsuitable here. 

One possibility is to run a simple script that will parse the finished `serving.yaml` manifest and spit out a `serving-gloo.yaml` with all Istio references removed.

Another is, rather than removing references to Istio from the default serving manifest, instead, simply pre-create the `istio-system` namespace and register Istio crds before applying the serving.yaml in the test setup.

Other than these two options, I'm not sure the best way to move forward with this part of the PR.

Reply:
a third option is to remove the alternate config path for Knative. always install Knative-Serving with Istio. Enabling Gloo for the e2e tests will not avoid deploying Istio, but instead simply ignore the fact that Istio is deployed. This will obviate the need to have an alternate deployment manifest for knative and remove all this duplication.

Reply:
@ilackarms after https://github.com/knative/serving/pull/2445 checks in you can install Knative without any Istio support by ignoring components with the label `networking.knative.dev/ingress-provider: istio`.